utils/parametric-components.stanza

```diff
@@ -332,6 +332,23 @@ public defn parametric-resistor (res:Double|Toleranced, tol:Double) -> Instantia
 ; ========== parametric part helper functions ============
 ; ========================================================
 
+public defn normalize-params (hs:Seqable<Seqable<KeyValue<String, ?>>>) -> Tuple<KeyValue<String, ?>> :
+  seq(substitute-params, hs)
+    $> remove-duplicate-keys
+
+; Substitute user params to match expectations:
+;   - https://en.wikipedia.org/wiki/Principle_of_least_astonishment
+defn substitute-params (params: Seqable<KeyValue<String, ?>>) -> Seqable<KeyValue<String, ?>> :
+  to-tuple $ for kv in params seq :
+    switch {key(kv) == _} :
+      "minimum_quantity" : "max-minimum_quantity" => value(kv)
+      "stock" : "min-stock" => value(kv)
+      else :
+        kv
```

Comment:
Maybe it is better to return things  as `Tuple` and use `Seqable` as argument

Reply:
Sounds good!

Reply:
Oh, it's already calling `to-tuple` above - the signature was just typed to `Seqable` - I'll update that to match `Tuple`.

Thanks!